### DISCUSSION TITLE
Using manifest mode and FIND_PACKAGE is not working

### DISCUSSION THREAD

**Discussion by Dariusz1989** [microsoft/vcpkg#29874] [Q&A]
Hello

I have this as my manifest>

```
{
  "name" : "c-dll-python-01",
  "version-string" : "1.0.0",
  "builtin-baseline" : "91dd61bd441a68b4017b61011d0350b2e6aeeccf",
  "dependencies" : [ {
    "name" : "python3"
  }, {
    "name" : "pybind11",
    "version>=" : "2.10.0#1"
  } ],
  "overrides" : [ {
    "name" : "python3",
    "version" : "3.7.3"
  } ]
}
```

I also have this in my CMAKE>
FIND_PACKAGE(Python3 3.7.3 EXACT COMPONENTS Interpreter Development REQUIRED)

He errors with this > 
```
  Could NOT find Python3 (missing: Python3_EXECUTABLE Interpreter) (found
  suitable exact version "3.7.3")

      Reason given by package: 
          Interpreter: Wrong version for the interpreter "C:/Program Files/Python37/python.exe"

```

How do I fix this?

**Comment by misuo** [comment]
Is the use of `overrides` the way to tell which version of Python you want to run embedded in pybind11? 

pybind11 v2.10.4 seems to be coming with Python 3.10 embedded, but how do you get Python 3.11 instead?

**Reply by Osyotr** [reply]
For python3.11 you create an overaly port based on this PR: https://github.com/microsoft/vcpkg/pull/31727

**Reply by misuo** [reply]
Thx for responding. I'm not familiar with overlay ports in conjunction with manifest mode.  I'm using vcpkg version 2023-06-15 and my manifest file `vcpkg.json` looks like:

```
{
  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg/master/scripts/vcpkg.schema.json",
  "name": "app",
  "version": "8.0.0",
  "dependencies": [
    "fmt",
    "restinio",
    "webview2",
    "dataframe",
    "pybind11"
  ]
```

The pybind11 retrieved is version 2.10.3 which uses embedded Python 3.10.7. So what do I have to do (write in manifest file) to get Python 3.11 embedded instead?

**Reply by Osyotr** [reply]
Create directory named `vcpkg_overlay_ports` (or simillar) near your `vcpkg.json`. Add `overlay-ports` entry to `vcpkg.json` or `vcpkg-configuration.json` (https://learn.microsoft.com/en-us/vcpkg/reference/vcpkg-json#example-3 or https://learn.microsoft.com/en-us/vcpkg/reference/vcpkg-configuration-json#example). Copy `python3` folder from https://github.com/microsoft/vcpkg/pull/31727 to `vcpkg_overlay_ports`.

**Comment by Osyotr** [comment]
Vcpkg's port for Python 3.7 is very old and simply didn't contain logic to build python interpreter. So if you really really want to use such an old version of python, you have to backport changes manually.
https://github.com/microsoft/vcpkg/pull/14891
